src/collection/account.service.ts

```diff
@@ -445,6 +447,49 @@ export class AccountService {
         expiredJwt.accountId.toString(),
       );
 
+      const timeUntilExpiration = expiredJwt.claims.exp - CURRENT_TIME_S;
+      const daysUntilExpiration = Math.floor(
+        timeUntilExpiration / (60 * 60 * 24),
+      );
+      if ([7, 5, 3, 2, 1].includes(daysUntilExpiration)) {
+        const emailSubject = `Token Expiration Warning for ${account.name}`;
+        const emailBody = `
+          Hello Broker Account ${account.name} Owner,
+  
+          This is a notification that your token associated with client ID ${
+            expiredJwt.claims.client_id
+          } will expire in ${daysUntilExpiration} day(s).
+  
+          Please take necessary actions to renew the token.
+  
+          Thank you,
+          OSCAR Team
+        `;
+        try {
+          await this.emailService.sendEmail(
+            account.email,
```

Comment:
Here's a problem... the account email is not validated and it is something we could look at dropping. So, chances are a lot of them are... questionable. There's also no guarantee it would go to anyone who could understand or action this. It needs to look for upstream lead-developers for their emails.